SimulationIOFilters/ImportDelamData.cpp

```diff
@@ -0,0 +1,568 @@
+#include "SimulationIO/SimulationIOVersion.h"
+
+enum createdPathID : RenameDataPath::DataID_t
+{
+  DataContainerID = 1,
+  AttributeMatrixID21 = 21,
+  DataArrayID30 = 30
+};
+
+// -----------------------------------------------------------------------------
+ImportDelamData::ImportDelamData()
+{
+  initialize();
+}
+
+// -----------------------------------------------------------------------------
+ImportDelamData::~ImportDelamData() = default;
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::initialize()
+{
+  clearErrorCode();
+  clearWarningCode();
+  setCancel(false);
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::setupFilterParameters()
+{
+  FilterParameterVectorType parameters;
+
+  parameters.push_back(SIMPL_NEW_INPUT_FILE_FP("CSDGM File", CSDGMFile, FilterParameter::Parameter, ImportDelamData, "*.txt"));
+  parameters.push_back(SIMPL_NEW_INPUT_FILE_FP("Bvid StdOut File", BvidStdOutFile, FilterParameter::Parameter, ImportDelamData, "*.txt"));
+  parameters.push_back(SIMPL_NEW_FLOAT_FP("Interface Thickness", InterfaceThickness, FilterParameter::Parameter, ImportDelamData));
+
+  parameters.push_back(SIMPL_NEW_DC_CREATION_FP("Data Container", DataContainerName, FilterParameter::CreatedArray, ImportDelamData));
+  parameters.push_back(SIMPL_NEW_STRING_FP("Cell Attribute Matrix", CellAttributeMatrixName, FilterParameter::CreatedArray, ImportDelamData));
+  parameters.push_back(SIMPL_NEW_STRING_FP("Data Array", DataArrayName, FilterParameter::CreatedArray, ImportDelamData));
+
+  setFilterParameters(parameters);
+}
+
+// -----------------------------------------------------------------------------
+void ImportDelamData::dataCheck()
+{
+  clearErrorCode();
+  clearWarningCode();
+
+  QFileInfo fi(m_CSDGMFile);
+  if(m_CSDGMFile.isEmpty())
+  {
+    QString ss = QObject::tr("Bvid File is empty.  Please enter a file path.");
+    setErrorCondition(-2000, ss);
+    return;
+  }
+
+  if(!fi.isFile() || fi.suffix() != "txt")
+  {
+    QString ss = QObject::tr("Bvid File is not a text file.  Please enter the path to a text file.");
+    setErrorCondition(-2001, ss);
+    return;
+  }
+
+  fi.setFile(m_BvidStdOutFile);
+  if(m_BvidStdOutFile.isEmpty())
+  {
+    QString ss = QObject::tr("Bvid StdOut File is empty.  Please enter a file path.");
+    setErrorCondition(-2002, ss);
+    return;
+  }
+
+  if(!fi.isFile() || fi.completeSuffix() != "txt")
+  {
+    QString ss = QObject::tr("Bvid StdOut File is not a text file.  Please enter the path to a text file.");
+    setErrorCondition(-2003, ss);
+    return;
+  }
+
+  readBvidStdOutFile();
+  if(getErrorCode() < 0)
+  {
+    return;
+  }
+
+  DataContainerShPtr dc = getDataContainerArray()->createNonPrereqDataContainer(this, getDataContainerName(), DataContainerID);
+
+  m_Xcrd = static_cast<int>(*m_Lcx / *m_Ex) + 1;
+  m_Ycrd = m_Xcrd;
+  m_Zcrd = 2 * *m_CzmLayers + 2;
```

Comment:
```suggestion
  m_Zcrd = 2 * (*m_CzmLayers) + 2;
```

The parentheses make this line a little easier to read though are not technically needed.